src/egon/data/datasets/power_plants/__init__.py

```diff
@@ -1014,48 +1044,43 @@ def fill_missing_bus_and_geom(gens, carrier):
         )
     ]
 
-    # convert from KW to MW
-    conv["Nettonennleistung"] = conv["Nettonennleistung"] / 1000
-    # drop generators installed after 2019
-    conv["Inbetriebnahmedatum"] = pd.to_datetime(conv["Inbetriebnahmedatum"])
-    conv = conv[
-        conv["Inbetriebnahmedatum"]
-        < egon.data.config.datasets()["mastr_new"]["status2019_date_max"]
-    ]
-
-    # drop chp generators
-    conv["ThermischeNutzleistung"] = conv["ThermischeNutzleistung"].fillna(0)
-    conv = conv[conv.ThermischeNutzleistung == 0]
-
-    # rename carriers
-    conv.loc[conv.Energietraeger == "Braunkohle", "Energietraeger"] = "lignite"
-    conv.loc[conv.Energietraeger == "Steinkohle", "Energietraeger"] = "coal"
-    conv.loc[conv.Energietraeger == "Erdgas", "Energietraeger"] = "gas"
-    conv.loc[
-        conv.Energietraeger == "Mineralölprodukte", "Energietraeger"
-    ] = "oil"
-    conv.loc[
-        conv.Energietraeger == "Kernenergie", "Energietraeger"
-    ] = "nuclear"
-
-    # rename columns
     conv.rename(
         columns={
+            "Inbetriebnahmedatum": "commissioning_date",
+            "EinheitBetriebsstatus": "status",
+            "DatumEndgueltigeStilllegung": "decommissioning_date",
             "EinheitMastrNummer": "gens_id",
             "Energietraeger": "carrier",
             "Nettonennleistung": "capacity",
             "Gemeinde": "location",
         },
         inplace=True,
     )
+
+    conv = discard_not_available_generators(conv, scenario_date_max)
+
+    # convert from KW to MW
+    conv["capacity"] = conv["capacity"] / 1000
+
+    # drop chp generators
+    conv["ThermischeNutzleistung"] = conv["ThermischeNutzleistung"].fillna(0)
+    conv = conv[conv.ThermischeNutzleistung == 0]
+
+    # rename carriers
+    conv.loc[conv.carrier == "Braunkohle", "carrier"] = "lignite"
```

Comment:
A dictionary plus [`pandas.Series.map`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.map.html) would be cleaner

Reply:
Implemented in 22b275de6172fb7f60c71459a2266a2be0e76b18